[JetNews] Use derived state of for scrolled state to avoid recomposition on scrolling

oas004 opened the pull request:
This should fix  #841  😃

bentrengrove reviewed (changes requested):
Thank you for this! This is a great idea.

This fixes it for this screen but `isScrolled` is used in multiple places in the app. A better fix would be moving the derivedStateOf to `LazyListUtils.kt`.

e.g

```
val LazyListState.isScrolled: Boolean
    get() = derivedStateOf { firstVisibleItemIndex > 0 || firstVisibleItemScrollOffset > 0 }.value
```

Could you make that change please?

oas004 commented:
> Thank you for this! This is a great idea.
> 
> This fixes it for this screen but `isScrolled` is used in multiple places in the app. A better fix would be moving the derivedStateOf to `LazyListUtils.kt`.
> 
> e.g
> 
> ```
> val LazyListState.isScrolled: Boolean
>     get() = derivedStateOf { firstVisibleItemIndex > 0 || firstVisibleItemScrollOffset > 0 }.value
> ```
> 
> Could you make that change please?

Thanks for review! Very good point! Changed it to use `derivedStateOf` in the `LazyListUtils` instead and rebased the commit 😄

bentrengrove reviewed (approved):
LGTM! Thanks for the contribution!
